pkg/execution/executor/validate.go

```diff
@@ -0,0 +1,176 @@
+package executor
+
+import (
+	"context"
+	"time"
+
+	"github.com/inngest/inngest/pkg/enums"
+	"github.com/inngest/inngest/pkg/execution"
+	"github.com/inngest/inngest/pkg/execution/queue"
+	"github.com/inngest/inngest/pkg/execution/state"
+	"github.com/inngest/inngest/pkg/inngest"
+	"github.com/inngest/inngest/pkg/logger"
+	"github.com/oklog/ulid/v2"
+)
+
+func newRunValidator(item queue.Item, s state.State, f *inngest.Function, e *executor) *runValidator {
+	return &runValidator{
+		item: item,
+		s:    s,
+		md:   s.Metadata(),
+		f:    f,
+		e:    e,
+	}
+}
+
+// runValidator runs checks when executing a queue item for the executor.
+type runValidator struct {
+	item queue.Item
+	s    state.State
+	md   state.Metadata
+	f    *inngest.Function
+
+	// stopWithoutRetry prevents
+	stopWithoutRetry bool
+
+	e *executor
+}
+
+func (r *runValidator) validate(ctx context.Context) error {
+	chain := []func(ctx context.Context) error{
+		r.checkCancelled,
+		r.checkStepLimit,
+		r.checkCancellation,
+		r.checkStartTimeout,
+		r.checkFinishTimeout,
+		r.updateScheduledStatus,
+	}
```

Comment:
literally just all of the `if` guards in the executor now as a pipeline.  if any guard errors, we'll return that error and not execute the step.